File: codeforces/cf1509B_tmt_document.py
# nel_tu_
# https://codeforces.com/contest/1509/problem/B
# greedy
def valid(n, s):
    if n % 3: return False

    count = [0, 0]
    T, M = 0, 1
    max_diff = n // 3

    for c in s:
        count[c == 'M'] += 1
        if count[M] > count[T]:
            return False
        if count[T] - count[M] > max_diff:
            return False

    return count[T] == 2 * count[M]

for _ in range(int(input())):
    n, s = int(input()), input()
    print('YES' if valid(n, s) else 'NO')

File: codeforces/cf1509B_tmt_document.hs
-- user: nel_tu_
-- path: https://codeforces.com/contest/1509/problem/B
-- tags: math, greedy

{-# LANGUAGE ViewPatterns #-}
import Control.Monad ( replicateM_ )
import Data.Bool ( bool )

solve :: Int -> String -> Bool
solve ((`div` 3) -> k) str = valid && t == k
    where
    (t, valid) = foldr loop (0, True) str

    loop x (upd x -> t, valid) = (t, valid && 0<=t && t<=k)

    upd 'T' = succ
    upd 'M' = pred


getInt = read <$> getLine
main = do
    n <- getInt
    replicateM_ n (solve <$> getInt <*> getLine  >>=  putStrLn . bool "NO" "YES")

